src/ray/core_worker/reference_count.h

```diff
@@ -92,12 +117,24 @@ class ReferenceCounter {
                          const std::function<void(const ObjectID &)> callback)
       LOCKS_EXCLUDED(mutex_);
 
+  /// Set a callback to reply to the given RPC when a borrower's ref count goes
+  /// to 0. The sender is the owner of the object ID. The borrower will send
+  /// the reply when its RefCount() for the object ID goes to 0.
+  ///
+  /// \param[in] request The request from the object's owner.
+  /// \param[in] reply A reply sent to the owner when we are no longer
+  /// borrowing the object ID. This reply also includes any new borrowers and
+  /// any object IDs that were nested inside the object that we or others are
+  /// now borrowing.
+  /// \param[in] send_reply_callback The callback to send the reply.
+  void SetRefRemovedCallback(const rpc::WaitForRefRemovedRequest &request,
```

Comment:
Do we want to expose the rpc messages here? Seems a bit messy - might be better to capture them in a callback?